commit 9a9da0f732005c2f954678aa8ab52241cd3e7cb4 (refs/changes/16/3216/2)
Author: Pedro Palazon Candel <pedro@joyent.com>
Date:   2018-01-16T14:36:53+01:00 (1 year, 9 months ago)
    
    TOOLS-1974 sdc-scripts update sdc_common_setup to allow creation of additional sapi instances

diff --git a/lib/util.sh b/lib/util.sh
index 8e07285..9ab798c 100644
--- a/lib/util.sh
+++ b/lib/util.sh
@@ -696,10 +696,17 @@ function sdc_common_setup
 
     if [[ ! -f /var/svc/setup_complete ]]; then
         if [[ ${ZONE_ROLE} != "assets" ]]; then
-            if [[ ${ZONE_ROLE} == "sapi" && "${SAPI_PROTO_MODE}" == \
-              "true" ]]; then
-                echo "Skipping config-agent/SAPI instance setup: 'sapi' " \
-                  "zone in proto mode" >&2
+            if [[ ${ZONE_ROLE} == "sapi" ]]; then
+                if [[ "${SAPI_PROTO_MODE}" == "true" ]]; then
+                    echo "Skipping config-agent/SAPI instance setup: 'sapi' " \
+                    "zone in proto mode" >&2
+                else
+                    echo "Setup config-agent and registrar services" >&2
+                    setup_config_agent
+                    download_metadata
+                    write_initial_config
+                    registrar_setup
+                fi
             else
                 setup_config_agent
                 download_metadata
